### PR TITLE
Add feature RBSFXIRJTP-208 -add relation between announce interest an…

### PR DESCRIPTION
**Pull request by radevdetelin**
…d changes in product database

update announceInterests in AuroraClientSideService to fit the requirements of the business logic, add announceInterestToOrderBookProductBase method to ProductServiceImpl to initiate interest when new product is added and relevant code to deleteProductFromDatabase method to act when a product is deleted

### CODE REVIEW

**Review by valto2 — APPROVED** [review]

File: calculator/src/main/java/com/market/banica/calculator/service/ProductServiceImpl.java [hunk 1]
```diff
@@ -145,6 +147,8 @@ public void deleteProductFromDatabase(String productName) {
 
         productBase.getDatabase().remove(productName);
 
+        auroraClientSideService.announceInterests(productName);
```
Comment (gaantonova) [inline calculator/src/main/java/com/market/banica/calculator/service/ProductServiceImpl.java:150]: Why do we subscribe to this item if we are deleting it from the DB?

Comment (radevdetelin) [reply calculator/src/main/java/com/market/banica/calculator/service/ProductServiceImpl.java:150]: announce interest method has a role to update a product base in order book, where only existing products are populated with new data from the market, so it has two roles in calculator service when passing changes to order book - to update with new product and to remove existing that are deleted, so no longer info is requested for these products

Comment (gaantonova) [reply calculator/src/main/java/com/market/banica/calculator/service/ProductServiceImpl.java:150]: using one method for subscribing and unsubscribing to an item can lead to a lot of confusion and to potential bugs, this is not a good architectural approach, please add another method that handles unsubscribing

Comment (radevdetelin) [reply calculator/src/main/java/com/market/banica/calculator/service/ProductServiceImpl.java]: Done

**Review by zaraandreevaa — APPROVED** [review]

**Pull request merged**